### PR TITLE
fix(Windows): fixed Windows OS file system path issue

### PR DESCRIPTION
**Pull request by stephenc222**
https://github.com/dcbarrett/vscode-coffeescript-preview/issues/3

This PR fixes the Windows OS file path generation issue above.

### CODE REVIEW

**Comment by dcbarrett** [comment]
@stephenc222 Thanks for submitting a PR for the issue. The change looks good to me, but a test should be added to cover the scenario. The generatePreviewUri function is tested indirectly in the updateContent function test [here](https://github.com/dcbarrett/vscode-coffeescript-preview/blob/master/test/coffeescript-preview.test.ts#L51). We may be able to inspect the URI object to ensure to appropriate separator is being used. Do you want to add the test?

**Review by dcbarrett — CHANGES_REQUESTED** [review]
Let's add a test for the scenario

**Comment by stephenc222** [comment]
Definitely! I'll add the test and update the PR. Thanks for the feedback!

File: test/coffeescript-preview.test.ts [hunk 1]
```diff
@@ -51,6 +52,11 @@ describe("CoffeeScriptPreview Tests", () => {
     it("updateContent() will call the content provider to update content", ()=>{
         const stub = sinon.stub(this.provider, "updateContent");
         this.cspreview.updateContent("new_file.coffee");
+        let separator = "//";
+        if (os.platform() === "win32") {
+            separator = "\\";
+        }
+        assert.equal(stub.getCall(0).args[0].toString().includes(separator), true);
```
Comment (dcbarrett) [inline test/coffeescript-preview.test.ts:59]: This looks like a great start. I'm thinking we should divide the test into two and test both scenarios (Windows and non-Windows). I'll go ahead and merge as is and see about adding it in.

**Review by dcbarrett — APPROVED** [review]

**Comment by dcbarrett** [comment]
Thanks for adding the test. I'll see about getting this deployed soon. I just welcomed a new addition to the family so it may be a few days, but I'll try to get to it asap.

**Comment by stephenc222** [comment]
Congratulations! Happy to help make an awesome vsc extension a tiny bit better!

**Pull request merged**